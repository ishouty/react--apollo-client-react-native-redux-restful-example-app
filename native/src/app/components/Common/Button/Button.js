import React, { Component } from "react"
import { Ionicons } from "@expo/vector-icons"
import { Text, View, StyleSheet, TouchableHighlight } from "react-native"
import Colors from "../../../constants/Colors"
import Fonts from "../../../constants/Fonts"

export default class Button extends Component {
  render() {
    const { title, callback, bsClass, shouldShowRightIcon } = this.props

    let buttonTitleStyle = shouldShowRightIcon ? styles.buttonTextLeft : styles.buttonTextCenter

    return (
      <View>
        <TouchableHighlight
          onPress={() => {
            callback()
          }}
        >
          <View style={[styles.buttonContainer, bsClass.button]}>
            <Text style={[buttonTitleStyle, bsClass.buttonText]}>{title}</Text>
          </View>
        </TouchableHighlight>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  buttonContainer: {
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    padding: 5,
    borderRadius: 0,
    borderColor: Colors.black,
    borderWidth: 2,
    height: 40
  },
  button: {
    borderRadius: 0,
    borderColor: Colors.black,
    borderWidth: 2,
    height: 40
  },
  buttonTextCenter: {
    padding: 5,
    color: Colors.black,
    fontFamily: Fonts.montserratBold,
    fontSize: 16,
    textAlign: 'center'
},
buttonTextLeft: {
    color: Colors.black,
    fontFamily: Fonts.montserratBold,
    fontSize: 16,
    flexGrow: 2
  },
  iconRight: {
    margin: 10
  }
})
