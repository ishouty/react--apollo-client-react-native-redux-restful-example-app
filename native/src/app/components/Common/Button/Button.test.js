import 'react-native'
import React from 'react'
import Button from './Button'
import renderer from 'react-test-renderer'

it('renders component correctly', () => {
    const tree = renderer.create(<Button title={'Test Button'} callback={()=> { }} bsClass={{button: {marginTop: 5}}} />).toJSON();

    expect(tree).toMatchSnapshot();
});
