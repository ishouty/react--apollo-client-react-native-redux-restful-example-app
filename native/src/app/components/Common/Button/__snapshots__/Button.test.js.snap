// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component correctly 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderColor": "#606060",
            "borderRadius": 0,
            "borderWidth": 2,
            "flexDirection": "row",
            "height": 40,
            "justifyContent": "center",
            "padding": 5,
          },
          Object {
            "marginTop": 5,
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#606060",
              "fontFamily": "montserrat-bold",
              "fontSize": 16,
              "padding": 5,
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Test Button
      </Text>
    </View>
  </View>
</View>
`;
